### PR TITLE
Move user command status columns to blocks_user_commands

### PR DESCRIPTION
**Pull request by omerzach**
- Updates archive DB schema to move columns
- Updates Archive_lib.Processor so the Transaction_status.t object is
  passed to Block_and_signed_command insertion instead of to
  User_command insertion
- Updates Rosetta Block code to handle the new schema when retrieving
  these columns
- Temporarily removes code in missing_subchain.ml that converted
  Archive_lib.Processor user commands to Extensional user commands since
  they'll now require an additional query

Explain how you tested your changes here.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them:

### CODE REVIEW

File: src/app/rosetta/lib/block.ml [hunk 1]
```diff
@@ -287,7 +287,16 @@ WITH RECURSIVE chain AS (
 
   module User_commands = struct
     module Extras = struct
-      type t = {fee_payer: string; source: string; receiver: string}
+      (* TODO: A few of these actually aren't used; should we leave in for future or remove? *)
```
Comment (omerzach) [inline src/app/rosetta/lib/block.ml:290]: We could remove `status` and `created_token` entirely from this file now as they're not used

**Comment by omerzach** [comment]
Context: https://github.com/MinaProtocol/mina/pull/7114#issuecomment-764814241

This is just the first bit to move the existing columns. I still need to actually add the new balance columns.

**Comment by mrmr1993** [comment]
I think this effectively disables some rosetta functionality for the time being. I have some concerns over merging this in isolation, @bkase what do you think?

The changes themselves look good though!

**Comment by bkase** [comment]
Yes it’s okay as long as we don’t break the ledger (so the bits of the construction flow needed to send signed payments — this would include getting the nonce) — I don’t have time to  check too closely now but as long as that’s the case, then we should be good to merge this.

**Review by mrmr1993 — APPROVED** [review]

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Pull request merged**